src/CTA.Rules.Update/ActionsRewriter.cs

```diff
@@ -321,11 +321,12 @@ public override SyntaxNode VisitCompilationUnit(CompilationUnitSyntax node)
         }
         public override SyntaxNode VisitObjectCreationExpression(ObjectCreationExpressionSyntax node)
         {
-            ObjectCreationExpressionSyntax newNode = node;// (ObjectCreationExpressionSyntax)base.VisitObjectCreationExpression(node);
-            bool skipChildren = false;
+            var symbols = _semanticModel.GetSymbolInfo(node);
+            ExpressionSyntax newNode = node;// (ObjectCreationExpressionSyntax)base.VisitObjectCreationExpression(node);
+            bool skipChildren = false; // This is here to skip actions on children node when the main dientifier was changed. Just use new expression for the subsequent children actions.
             foreach (var action in _fileActions.ObjectCreationExpressionActions)
             {
-                if (newNode.ToString() == action.Key)
+                if (newNode.ToString() == action.Key || symbols.Symbol.ContainingType.Name == action.Key)
```

Comment:
Need to add a null check here (in case symbol is null)

Reply:
Thanks Mark, good catch, I went ahead and used the '?' character after symbols.Symbol? to force a null return to avoid have a null reference exception and I added one to symbols.Symbol?.ContainingType? as well to be safe.